src/odemis/driver/hamamatsurx.py

```diff
@@ -0,0 +1,2116 @@
+        return md
+
+    def _getDataFromBuffer(self):
+        """
+        This method runs in a separate thread and waits for messages in queue indicating
+        that some data was received. The image is then received from the device via the dataport IP socket or
+        the vertical scaling table is received, which corresponds to a time range for a single sweep.
+        It corrects the vertical time information. The table contains the actual timestamps for each px..
+        """
+
+        # TODO need to check that there is an ringbuffer available!?
+
+        logging.debug("Starting data thread.")
+        time.sleep(1)
+        is_receiving_image = False
+
+        try:
+            while True:
+
+                if self._sync_event and not is_receiving_image:
+                    while int(self.parent.AsyncCommandStatus()[0]):
+                        time.sleep(0)
+                        logging.debug("Asynchronous RemoteEx command still in process. Wait until finished.")
+                        # TODO if not finished after some time might be live mode, so StopAcq again
+                    try:
+                        event_time = self.queue_events.popleft()
+                        logging.warning("Starting acquisition delayed by %g s.", time.time() - event_time)
+                        self.parent.AcqStart(self.acqMode)
+                        is_receiving_image = True
+                    except IndexError:
+                        # No event (yet) => fine
+                        pass
+
+                rargs = self.parent.queue_img.get(block=True)  # block until receive something
+                logging.debug("Received message %s", rargs)
+
+                if rargs is None:  # if message is None end the thread
+                    return
+
+                # synchronized mode
+                if self._sync_event:
+                    if rargs == "start":
+                        logging.info("Received event trigger")
+                        continue
+                    else:
+                        logging.info("Get the synchronized image.")
+
+                # non-sync mode
+                else:
+                    while not self.parent.queue_img.empty():
+                        # keep reading to check if there might be a newer image for display
+                        # in case we are too slow with reading
+                        rargs = self.parent.queue_img.get(block=False)
+
+                        if rargs is None:  # if message is None end the thread
+                            return
+                    logging.info("No more images in queue, so get the image.")
+
+                if rargs == "F":  # Flush => the previous images are from the previous acquisition
+                    logging.debug("Acquisiton was stopped so flush previous images.")
+                    continue
+
+                self._metadata[model.MD_ACQ_DATE] = time.time() - \
```

Comment:
Currently, the streakccd has a metadata MD_ACQ_DATE, which makes little sense on the detector itself. For this metadata, IMHO, it'd make more sense to only set the copy of the metadata => move after l. 525.
To ensure you still have a good precision, you could just store the time here with `reception_time = time.time()`, and l.526 do something like:
`md[model.MD_ACQ_DATE] = reception_time - self.exposureTime.value + md[model.MD_READOUT_TIME]`

BTW, normally, the CCD drivers in Odemis also provide a MD_EXP_TIME on the image metadata. It should be set whenever you change the exposureTime. If you add it, be careful to check that your acquisition code handles it properly. In particular, there shouldn't be both a MD_DWELL_TIME (from the e-beam) and a MD_EXP_TIME (from the CCD). So the code needs to pick one.